tests/new_upgrades/test_puppet.py

```diff
@@ -0,0 +1,97 @@
+"""Test for Remote Execution related Upgrade Scenario's
+
+:Requirement: UpgradedSatellite
+
+:CaseAutomation: Automated
+
+:CaseComponent: Puppet
+
+:Team: Rocket
+
+:CaseImportance: Medium
+
+"""
+
+from box import Box
+import pytest
+
+from robottelo.config import settings
+from robottelo.utils.shared_resource import SharedResource
+
+
+@pytest.fixture
+def puppet_upgrade_setup(puppet_upgrade_integrated_sat_cap, upgrade_action):
+    satellite = puppet_upgrade_integrated_sat_cap.satellite
+    capsule = puppet_upgrade_integrated_sat_cap.capsule
+    with (
+        SharedResource(satellite.hostname, upgrade_action, target_sat=satellite) as sat_upgrade,
+        SharedResource(capsule.hostname, upgrade_action, target_sat=capsule) as cap_upgrade,
+    ):
+        test_data = Box(
+            {
+                'satellite': satellite,
+                'capsule': capsule,
+            }
+        )
+        sat_upgrade.ready()
+        satellite._swap_nailgun(f'{settings.UPGRADE.TO_VERSION}.z')
+        cap_upgrade.ready()
+        satellite._session = None
+        capsule._session = None
+        yield test_data
+
+
+@pytest.mark.capsule_upgrades
```

Comment:
Should we also use `pytest.mark.puppet_upgrades` marker for the test ?

Reply:
Good catch, yes. I'll make the change.